File: src/main/java/co/uk/isxander/lunarspoof/websocket/auth/packet/impl/SPacketAuthenticatedRequest.java
package co.uk.isxander.lunarspoof.websocket.auth.packet.impl;

import co.uk.isxander.lunarspoof.websocket.auth.LunarAuthWebSocket;
import co.uk.isxander.lunarspoof.websocket.auth.packet.LunarPacket;
import co.uk.isxander.xanderlib.utils.json.BetterJsonObject;

public class SPacketAuthenticatedRequest extends LunarPacket {

    private String jwtKey;

    public String getJwtKey() {
        return this.jwtKey;
    }

    @Override
    public String getName() {
        return "SPacketAuthenticatedRequest";
    }

    @Override
    public void process(LunarAuthWebSocket ws) {
        ws.acceptAuthentication(this);
    }

    @Override
    public void processJson(BetterJsonObject json) {
        this.jwtKey = json.optString("jwtKey");
    }

}

File: src/main/kotlin/dev/isxander/lunarspoof/websocket/auth/packet/impl/SPacketAuthenticatedRequest.kt
package dev.isxander.lunarspoof.websocket.auth.packet.impl

import dev.isxander.lunarspoof.websocket.auth.LunarAuthWebSocket
import dev.isxander.lunarspoof.websocket.auth.packet.AuthenticatorPacket
import gg.essential.api.utils.JsonHolder

class SPacketAuthenticatedRequest : AuthenticatorPacket() {
    override val name: String = "SPacketAuthenticatedRequest"
    lateinit var jwtKey: String
        private set

    override fun process(ws: LunarAuthWebSocket) {
        ws.acceptAuthentication(this)
    }

    override fun processJson(json: JsonHolder) {
        this.jwtKey = json.optString("jwtKey")
    }
}